src/extensions/client/mod.rs

```diff
@@ -261,13 +261,17 @@ impl Client {
                 }
                 // wait for at least one endpoint to connect
                 futures::future::select_all(endpoints.iter().map(|x| x.connected().boxed())).await;
-                // Sort by health score
-                endpoints.sort_by_key(|endpoint| std::cmp::Reverse(endpoint.health().score()));
-                // Pick the first one
-                endpoints[0].clone()
+
+                endpoints
+                    .iter()
+                    .max_by_key(|endpoint| endpoint.health().score())
+                    .expect("No endpoints")
+                    .clone()
             };
 
-            let mut selected_endpoint = healthiest_endpoint(None).await;
```

Comment:
this is important. it ensures at least one endpoint is connected. selecting just the first one may result on endpoint connection failure and never connects so `selected_endpoint.connected().await` will never resolve

Reply:
in that case we need a test. the current behaviour makes unit test non-deterministic as it may connect any of the dummy server